server/index.js

```diff
@@ -95,6 +95,9 @@ global.LoginServer = require('../.server-dist/loginserver').LoginServer;
 global.Ladders = require('./ladders');
 
 global.Chat = require('./chat');
+Chat.version().then(function (hash) {
```

Comment:
This sounds like a race condition in the making?

Maybe you should do something like

```suggestion
global.__verison = {};
Chat.version().then(function (hash) {
```